code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -84,6 +84,47 @@
 		if(standard_pour_into(user, target))
 			return
 
+		//Help the user avoid clumsy splashing mistakes due to finicky sprites etc.
+		if (reagents.total_volume) //don't pop a dialog if it's empty...
+			//handle lidded containers
+			if  (user.clumsy == 0 && (istype(target, /obj/item/weapon/reagent_containers)))
+				var/resp = alert(user, "Do you really want to splash the solution?", "Splash the solution?", "Abort", "Proceed")
+				if (resp == "Abort")
+					return
+				else if (!user.Adjacent(target))
+					user << "You cannot reach the [src]."
+					return
+				else if (user.stat != 0)
```

Comment:
@Iamgoofball The problem lies in the proc being called and the user dying while the dialog is open, allowing liquids to be poured beyond the veil, see Kelenius' previous comments.